### PR TITLE
Add fix to pivot destination number

### PR DESCRIPTION
**Pull request by avalonche**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
-

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [ ] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs [hunk 1]
```diff
@@ -111,32 +133,29 @@ protected override AddBlockResult InsertToBlockTree(BlockHeader header)
...
-            // if (_blockTree.LowestInsertedHeader != null
-            //     && _blockTree.LowestInsertedHeader.Number < (_blockTree.LowestInsertedBeaconHeader?.Number ?? long.MaxValue))
-            // {
-            if (_logger.IsTrace)
-                _logger.Trace(
-                    " BeaconHeader LowestInsertedBeaconHeader found existing chain in fast sync," +
-                    $"old: {_blockTree.LowestInsertedBeaconHeader?.Number}, new: {_blockTree.LowestInsertedHeader.Number}");
-            // beacon header set to (global) lowest inserted header
-            //   _blockTree.LowestInsertedBeaconHeader = _blockTree.LowestInsertedHeader;
+            if (_blockTree.LowestInsertedBeaconHeader?.ParentHash == header.Hash)
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs:140]: I don't understand, if the header feed feed the header in reverse (head to root), then shouldn't the previous insterted block here always be the LowestInsertedBeaconHeader, and it's parent always be the current `header` and therefore, this block will always be true?

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconPivot.cs [hunk 2]
```diff
@@ -75,7 +75,6 @@ public class BeaconPivot : IBeaconPivot
 
         public long PivotDestinationNumber => CurrentBeaconPivot is null
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconPivot.cs:76]: PivotDestinationNumber is used only for logging? Do we have correct logging?

Comment (avalonche) [reply src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconPivot.cs:76]: it is not 100% accurate as the destination number moves as blocks gets processed from genesis in archive sync. However it is good enough for the sync reporter

**Review by MarekM25 — CHANGES_REQUESTED** [review]
New hive tests are failing: https://github.com/NethermindEth/nethermind/runs/7080452606?check_suite_focus=true

**Pull request closed without merging**